### PR TITLE
chore(forms): fix proptypes and lint errors

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

`✖ 131 problems (131 errors, 0 warnings)`

**What is the chosen solution to this problem?**

`✖ 5 problems (5 errors, 0 warnings)`

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -7,120 +7,32 @@ Scoping to packages that match 'react-talend-forms'
...
-  544:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
-  552:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
-  553:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
-  556:3   error  Prop type `array` is forbidden                                                                 react/forbid-prop-types
+  195:32  error  Unexpected block statement surrounding arrow body                                              arrow-body-style
+  207:40  error  Unexpected block statement surrounding arrow body                                              arrow-body-style
+  225:32  error  Unexpected block statement surrounding arrow body                                              arrow-body-style
+  241:16  error  Expected 'this' to be used by class method 'isItemRequired'                                    class-methods-use-this
```
Comment (jmfrancois) [inline output/forms.eslint.txt]: easy to fix ☝️

File: output/forms.eslint.txt [hunk 2]
```diff
@@ -7,120 +7,32 @@ Scoping to packages that match 'react-talend-forms'
...
+  560:3   error  Prop type `array` is forbidden                                                                 react/forbid-prop-types
+  566:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  579:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  586:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  613:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  614:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  634:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
+  635:3   error  Prop type `object` is forbidden                                                                react/forbid-prop-types
```
Comment (jmfrancois) [inline output/forms.eslint.txt]: please update eslintrc with the new one from tools

File: output/forms.eslint.txt (hunk 1 above)
Comment (jsomsanith) [inline output/forms.eslint.txt]: Done

File: output/forms.eslint.txt (hunk 2 above)
Comment (jsomsanith) [inline output/forms.eslint.txt]: Let's merge this one, and make another PR to update the eslintrc + remove the suppress `react/forbid-prop-types` in all files
